src/Lecture1.hs

```diff
@@ -53,8 +54,8 @@ their squares.
 Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
 is 25.
 -}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+sumOfSquares :: Int -> Int -> Int
+sumOfSquares x y = x^2 + y^2
```

Comment:
This is the correct solution 👍🏻 I see that the Haskell compiler outputs some warnings here. This is because the operator `^` is polymorphic on both arguments and numeric literals are polymorphic as well. 

This is unfortunate 😞 But it's a problem when using the `^` operator. One of the possible solutions is to specify the type of both constants `2` to some specific type using type annotations:

```suggestion
sumOfSquares x y = x ^ (2 :: Int) + y ^ (2 :: Int)
```

I agree that it looks a bit ugly now 😕 

Another approach is to use `*` in this particular case to avoid ambiguity:

```suggestion
sumOfSquares x y = x * x + y * y
```